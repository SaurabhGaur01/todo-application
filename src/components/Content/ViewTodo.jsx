import React from 'react';
import PropTypes from 'prop-types';
import Checkbox from '@material-ui/core/Checkbox';

/*
This component is responsible for displaying todo item with checkbox and label
*/
const ViewTodo = ({ 
  id, 
  onClick, 
  completed, 
  text 
}) => (
  <li key={id}>
    <Checkbox onClick={onClick} checked={completed} label={(
      <label style={{textDecoration: completed ? 'line-through' : 'none'}}>
        {text}
      </label>
    )}/>
  </li>
);

ViewTodo.propTypes = {
  id: PropTypes.number.isRequired,  
  onClick: PropTypes.func.isRequired,
  completed: PropTypes.bool.isRequired,
  text: PropTypes.string.isRequired
}

export default ViewTodo;