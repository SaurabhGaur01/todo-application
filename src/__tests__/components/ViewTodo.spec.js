import React from 'react';
import { shallow } from 'enzyme';
import ViewTodo from '../../components/Content/ViewTodo';

describe('ViewTodo Component', () => {
    let props;

    beforeEach(() =>{
        props = {
            id: 1,
            completed: true,
            text: 'mockText',
            onClick: jest.fn(),
        }
    });

    it('should render correctly', () => {
        const renderedModule = shallow(<ViewTodo {...props} />);
        expect(renderedModule).toMatchSnapshot();
    });
})