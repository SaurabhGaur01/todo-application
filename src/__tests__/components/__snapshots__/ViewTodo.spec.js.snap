// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewTodo Component should render correctly 1`] = `
<li
  key="1"
>
  <WithStyles(ForwardRef(Checkbox))
    checked={true}
    label={
      <label
        style={
          Object {
            "textDecoration": "line-through",
          }
        }
      >
        mockText
      </label>
    }
    onClick={[MockFunction]}
  />
</li>
`;
